File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if head is None:
            return False
        
        node = head
        
        while node:
            if hasattr(node, "has_been_visited"):
                return True
            else:
                node.has_been_visited = True
            node = node.next

File: 0141-linked-list-cycle/0141-linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let node = head;
    
    while (node) {
        if (node.wasSeen) {
            return true;
        } else {
            node.wasSeen = true;
        }
        node = node.next;
    }
    
    return false;
};